Improve playback recovery when dealing with faulty files.

mstratford opened the pull request:
In some odd cases, audio files can get corrupted.

In the case this is server side, there's not much we should/can do.

However, if this has occured as part of the BAPSicle normalisation or download process, we should reattempt.

This PR adds better recovery support when loading files, by switching back to the non-normalised file, or failing that, redownload the original from the API and play that.

mstratford committed:
Fix returning OKAY to play when unloaded. Closes #16

mstratford committed:
Try non-normalised / redownload file on failure to load.

mstratford committed:
Add better linting and 3.1.0

mstratford committed:
Little lint

mstratford commented:
I'm sure with enough effort the checks for existing files etc before going into the try loop could be removed, but heh, it works.

mstratford committed:
Try unique package names per build.

mstratford committed:
Try other refs.

mstratford committed:
Just pull request with sha.

mstratford committed:
Try getting branch via bash.

mstratford committed:
Get branch without user prefix (can't have / in filename)

michael-grace commented:
so, if it tries to download a track from myradio, and it times out, then it restarts the player cause it throws a timeout exception, and never sends the websocket message to load the thing, which just makes nothing happen, so maybe we put that case in to say "load failed"

mstratford commented:
> so, if it tries to download a track from myradio, and it times out, then it restarts the player cause it throws a timeout exception, and never sends the websocket message to load the thing, which just makes nothing happen, so maybe we put that case in to say "load failed"

I will try and replicate this, the player isn't supposed to restart. In my mind, what's supposed to happen is "self.isLoaded" goes false, but with the "loaded_item" being set, and presenter would then show a "Load failed" error.

That does bring up the "What if it's still loading", I think webstudio does show Load Failed for a brief sec till BAPSicle has updated to say it's loaded.

mstratford committed:
Merge remote-tracking branch 'origin/dev' into mstratford/reload-repair

mstratford committed:
Show loaded status

mstratford committed:
Fix borked API, makes webstudio not white screen.

mstratford committed:
404 correctly on missing audio files.

mstratford committed:
Fix a failed download leaving temp file.

mstratford committed:
Semi-fix load failed error in webstudio not showing

If a load failed immediately, webstudio didn't see anything loaded, so doesn't report it failed.

mstratford committed:
Fix preloader not preloading due to GET_PLAN change to GETPLAN

mstratford committed:
3.1.0 again?

mstratford committed:
Merge branch 'dev' into mstratford/reload-repair
